roles/validations/tasks/edpm/hugepages_and_reboot.yml

```diff
@@ -0,0 +1,128 @@
+# This job tests the functionality of the dataplane-operator to update values for a given
+# openstackdataplanenodeset and have them applied to nodes within the cluster. Specifically,
+# we will update the value for hugepages, ensure it is applied to the node, and that the node
+# has been rebooted.
+#
+# This job was created to satisfy:
+# https://issues.redhat.com/browse/osprh-5324
+
+# Collect facts from the node in question. this will give us an initial state prior to our changes
+# that can be compared with the end state.
+- name: Collect initial state of edpm node
+  ansible.builtin.gather_facts:
+  delegate_to: "{{ cifmw_validations_edpm_check_node }}"
+  register: initial_node_state
+
+# Since jobs use different names for their OpenStackDataPlaneNodeSet's, we need to dynamically
+# determine the name of the currently deployed one
+- name: Determine name of deployed NodeSet
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  cifmw.general.ci_script:
+    output_dir: "{{ cifmw_validations_basedir }}/artifacts"
+    script: >-
+      oc get osdpns --no-headers -o custom-columns=":metadata.name"
+  register: deployed_nodeset_name
+
+# collect initial confighash from the nodeset
+- name: Collecting initial confighash from the nodeset
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  cifmw.general.ci_script:
+    output_dir: "{{ cifmw_validations_basedir }}/artifacts"
+    script: >-
+      set -o pipefail && oc get -n {{ cifmw_install_yamls_defaults['NAMESPACE'] }} osdpns "{{ deployed_nodeset_name.stdout }}" -o yaml | yq .status.configHash
```

Comment:
@pablintino @rebtoor guess we shouldn't rely on `cifmw_install_yamls_defaults['NAMESPACE']` in an "architecture" deployment, should we?
I'm not sure the `openshift_login` exposes any namespace - is there any other trustful source for that data, or should we get yet another `cifmw_validations_namespace` parameter?

Any thoughts?

Reply:
Correct, if this is going to be used for VAs we cannot rely on that variable. Here I see two approaches:

- Add an extra var, like you suggest
- Assume that a deployment will have a single dataplane instance and fetch the NS that has that deployment beforehand.